setup.sh

```diff
@@ -0,0 +1,239 @@
+        break ;;
+      *)
+        docker run \
+          --name paradedb \
+          -e POSTGRES_USER="$pguser" \
+          -e POSTGRES_PASSWORD="$pgpass" \
+          -e POSTGRES_DB="$dbname" \
+          -p 5432:5432 \
+          -d \
+          paradedb/paradedb:latest || { echo "Failed to start Docker container. Please check if an existing container is active or not."; exit 1; }
+        break ;;
+    esac
+  done
+  echo "Docker Container started ✅"
+
+  # Provide usage information
+  echo -e "\n\nTo use paradedb execute the command: docker exec -it paradedb psql $dbname -U $pguser"
+}
+
+installDeb(){
+  echo "Select your distribution"
+
+  echo "Installing dependencies...."
+  echo "Installing cURL"
+
+  sudo apt-get update -y || false
+  sudo apt-get install curl -y || false
+
+  echo "Successfully Installed cURL✅"
+
+  distros=("bookworm(Debian 12.0)" "jammy(Ubuntu 22.04)" "noble(Ubuntu 24.04)")
+  distro=
+  select op in "${distros[@]}"
+  do
+    case $op in
+      "bookworm(Debian 12.0)")
+        distro="bookworm"
+        break ;;
+      "jammy(Ubuntu 22.04)")
+        distro="jammy"
+        break ;;
+      "noble(Ubuntu 24.04)")
+        distro="noble"
+        break ;;
+    esac
+  done
+
+  if [ "$ARCH" = "x86_64" ]; then
+    ARCH="amd64"
+  fi
+
+  filename="postgresql-$1-pg-search_${LATEST_RELEASE_VERSION}-1PARADEDB-${distro}_${ARCH}.deb"
+  url="https://github.com/paradedb/paradedb/releases/latest/download/${filename}"
+
+  echo "Downloading ${url}"
+
+  curl -L "$url" > "$filename" || false
+
+  sudo apt install ./"$filename" -y || false
+}
+
+# Installs latest RPM package
+installRPM(){
+  filename="pg_search_$1-$LATEST_RELEASE_VERSION-1PARADEDB.el9.${ARCH}.rpm"
+  url="https://github.com/paradedb/paradedb/releases/latest/download/${filename}"
+  echo -e "Installing cURL"
+  sudo dnf install curl || false
+  echo "Successfully Installed cURL✅"
+
+  echo "Downloading ${url}"
+  curl -l "$url" > "$filename" || false
+
+  sudo rpm -i "$filename" || false
+  echo "ParadeDB installed successfully!"
+}
+
+# Installs latest binary for ParadeDB
+installBinary(){
+
+  # Select postgres version
+  pg_version=
+  echo "Select postgres version(Note: ParadeDB is supported on PG12-16. For other postgres versions, you will need to compile from source.)"
+  versions=("14" "15" "16")
+
+  select vers in "${versions[@]}"
+  do
+    case $vers in
+      "14")
+        pg_version="14"
+        break ;;
+      "15")
+        pg_version="15"
+        break ;;
+      "16")
+        pg_version="16"
+        break ;;
+    esac
+  done
+
+  # Select Base type
+  echo "Select supported file type: "
+  opts=(".deb" ".rpm")
+
+  select opt in "${opts[@]}"
+  do
+    case $opt in
+      ".deb")
+        installDeb $pg_version
+        break ;;
+      ".rpm")
+        installRPM $pg_version
+        break ;;
+    esac
+  done
+}
+
+
+echo -e "=========================================================\n"
```

Comment:
We should add an OS check. The user shouldn't get suggested installing the Stable Binary if they're on macOS or Windows, since they're not supported. I just had this happen.